### PR TITLE
[DEV-1099] Yield Machine Withdrawals

### PR DESCRIPTION
**Pull request by davidleomay**

### CODE REVIEW

**Review by SwissBlueDragon — APPROVED** [review]

File: src/subdomains/staking/infrastructure/staking-defichain.service.ts [hunk 1]
```diff
@@ -142,4 +122,53 @@ export class StakingDeFiChainService {
...
+  }
+
+  private getPossibleWithdrawalsFor(balance: BigNumber, assetName: string, withdrawals: Withdrawal[]): Withdrawal[] {
+    const sortedWithdrawals = withdrawals.filter((w) => w.asset.name === assetName).sort((a, b) => a.amount - b.amount);
+
+    const possibleWithdrawals = [];
+    let withdrawalSum = 0;
+    for (const withdrawal of sortedWithdrawals) {
+      withdrawalSum += withdrawal.amount;
+      if (balance.minus(1).lt(withdrawalSum)) break;
```
Comment (Krysh90) [inline src/subdomains/staking/infrastructure/staking-defichain.service.ts:144]: 1 is min withdrawal size?

**Review by Krysh90 — APPROVED** [review]

File: src/subdomains/staking/infrastructure/staking-defichain.service.ts (hunk 1 above)
Comment (davidleomay) [reply src/subdomains/staking/infrastructure/staking-defichain.service.ts:144]: That's just a buffer, could probably be smaller 🤔

Comment (Krysh90) [reply src/subdomains/staking/infrastructure/staking-defichain.service.ts:144]: buffer for what? I don't quite understand why it is there, what should it prevent and why?

Comment (davidleomay) [reply src/subdomains/staking/infrastructure/staking-defichain.service.ts:144]: In the end it's just a fee buffer. If we have 100 DFI on the liquidity address, we won't be able to pay out 100 DFI.

**Pull request merged**